packages/jest-each/src/bind.ts

```diff
@@ -82,7 +82,8 @@ const applyArguments = <EachCallback extends Global.TestCallback>(
   supportsDone: boolean,
   params: Array<unknown>,
   test: Global.EachTestFn<EachCallback>,
+  index: number,
 ): Global.EachTestFn<any> =>
   supportsDone && params.length < test.length
-    ? (done: Global.DoneFn) => test(...params, done)
-    : () => test(...params);
+    ? (done: Global.DoneFn) => test(...params, done, index)
+    : () => test(...params, index);
```

Comment:
this looks like it'll be wrong if people want the `index`, but not the `done` callback.

@orta since you requested the PR, thoughts on how to handle it?

Reply:
> this looks like it'll be wrong if people want the index, but not the done callback.

A work-around can be used, such as `test(...params, _, index)`

Reply:
Hrm, yeah, I'm not wild on it - but I think if this includes docs changes mentioning it then it's an acceptable trade-off

Reply:
That means, I just need to add documentation for this part then we can proceed? or is there any functional changes required?

Reply:
Yeah, IMO, this needs docs changes and it should be good

Reply:
I don't think that really address it as I think this removes the possibility to use promises (as we do `fn.length` to see if we should inject the `done` callback - if the user doesn't call it the test will time out).

I might be wrong though. CI fails on a lot of `each` tests, but hard to say if it's because they assert arguments or are actually broken

Reply:
> CI fails on a lot of each tests

Yes, it's because they assert arguments

Reply:
![Screenshot 2021-05-25 at 3 18 08 PM](https://user-images.githubusercontent.com/43617894/119477254-6dbaac00-bd6c-11eb-83bd-3543450c1dba.png)

Have a look

Reply:
> think this removes the possibility to use promises

Then  I think this idea seems like a write-off to me - that's certainly not an acceptable trade-off

Reply:
> this removes the possibility to use promises (as we do fn.length to see if we should inject the done callback - if the user doesn't call it the test will time out).

we can improve the logic for comparing `fn.length`.